conan/tools/cmake/cmakedeps/cmakedeps.py

```diff
@@ -75,26 +79,39 @@ def content(self):
             if dep.is_build_context and dep.ref.name not in self.build_context_activated:
                 continue
 
+            # Skip from the consumer
+            if any(fnmatch(str(dep.ref), pattern) for pattern in self.excluded_requirement_patterns):
+                self._conanfile.output.info("CMakeDeps: Skipped '{}'".format(dep.ref))
+                continue
+
+            # Skip from the requirement
             if dep.new_cpp_info.get_property("skip_deps_file", "CMakeDeps"):
                 # Skip the generation of config files for this node, it will be located externally
                 continue
 
-            config_version = ConfigVersionTemplate(self, require, dep)
-            ret[config_version.filename] = config_version.render()
+            generate_find_module = dep.new_cpp_info.get_property("cmake_module_file_name",
+                                                                 "CMakeDeps") is not None
+
+            for find_module_mode in ([False, True] if generate_find_module else [False]):
```

Comment:
Are you sure about generating both the Config and the Find? What would be the use case? Shouldn't be a selection? Maybe for each pattern/dependency to be able to select: None (not generate anything), Find (find_package) or Config (default)?

Reply:
Yes. You have to generate both because that property would be declared in recipes of libraries that have a FindXXX in the CMake installation so there might be consumers using `find_package(XXX MODULE`, of course, there might be consumers calling `find_package(xxx)` expecting to find the author's config file (that will find the Conan's xxx-config.cmake). 
So, it is not a consumer choice but a recipe choice.

Reply:
But I agree we might use a single property with different values set to `None` (current skip_deps_file), `Config`, `Module`, `Both`.